recipes/mimalloc/all/conanfile.py

```diff
@@ -0,0 +1,80 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class MimallocConan(ConanFile):
+    name = "mimalloc"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/microsoft/mimalloc"
+    description = "mimalloc is a compact general purpose allocator with excellent performance."
+    topics = ("libraries", "c", "c++", "malloc", "allocator")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"fPIC": [True, False],
+               "shared": [True, False],
+               "override": [True, False]}
+    default_options = {"fPIC": True,
+                       "shared": False,
+                       "override": True}
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt", "patches/*"
+    _cmake_i = None
+
+    @property
+    def _cmake_install_folder(self):
+        version = tools.Version(self.version)
+        return "{}-{}.{}".format(self.name, version.major, version.minor)
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _cmake(self):
+        if not self._cmake_i:
+            self._cmake_i = CMake(self)
+            self._cmake_i.definitions["MI_BUILD_TESTS"] = False
+            self._cmake_i.definitions["MI_BUILD_SHARED"] = self.options.shared
+            self._cmake_i.definitions["MI_BUILD_STATIC"] = not self.options.shared
+            self._cmake_i.configure(build_folder=self._build_subfolder)
+        return self._cmake_i
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("mimalloc-" + self.version, self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, "14")
+        if self.options.shared:
+            del self.options.fPIC
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        self._cmake.build()
+
+    def package(self):
+        self._cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", self._cmake_install_folder, "cmake"))
+
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.includedirs = [os.path.join("lib", self._cmake_install_folder, "include")]
+        self.cpp_info.libdirs = [os.path.join("lib", self._cmake_install_folder)]
```

Comment:
This makes it work on gcc @ Linux.

```suggestion
        self.cpp_info.libdirs = [os.path.join("lib", self._cmake_install_folder)]
        objext = "obj" if self.settings.compiler == "Visual Studio" else "o"
        self.cpp_info.exelinkflags = [os.path.join(self.package_folder, "lib", self._cmake_install_folder, "mimalloc.{}".format(objext)]
        self.cpp_info.sharedlinkflags = [os.path.join(self.package_folder, "lib", self._cmake_install_folder, "mimalloc.{}".format(objext)]
```

Reply:
@jgsogo 
mimalloc installs a library with the bare minimum functionality + a object file with extra functionality.
I add this using `cpp_info.exelinkflags` and `cpp_info.sharedlinkflags`.
Would a `cpp_info.objects` property make sense?

Reply:
I don't know of `cpp_info.objects` actually, let me check it on the docs. The object being built (`mimalloc.o`) is related to the "override" mode, in which the whole executable is forced to link against mimalloc allocator...

I was thinking about adding a "override" option (actually already did it, but just the member without any logic). As I'm learning this library as I'm going I was thinking of making this option do "the right thing", but I'm only reading the docs now...

For instance:

> On Unix-like systems, you can also statically link with mimalloc to override the standard malloc interface. The recommended way is to link the final program with the mimalloc single object file (mimalloc-override.o).

(https://github.com/microsoft/mimalloc#static-override)

Reply:
Do you think this makes sense?

Reply:
If I read it correctly,
then the dynamic override will only works when building mimalloc as a shared library.

For a static override, you can choose between linking to the library or linking to the object file.
The docs say that linking to the object file is more reliable.
This makes me thinking that the static library and the object file are more or less equivalent.

This object file is the same for a shared and static mimalloc build.
So it should probably be removed when a shared library is built.


I think a [`secure` option](https://github.com/microsoft/mimalloc#secure-mode) would also make sense